src/k8s/api/v1/types.go

```diff
@@ -58,6 +71,32 @@ func BootstrapConfigFromMap(m map[string]string) (*BootstrapConfig, error) {
 	return config, nil
 }
 
+type JoinClusterConfig struct {
+	APIServerCert string `yaml:"apiserver-crt,omitempty"`
+	APIServerKey  string `yaml:"apiserver-key,omitempty"`
+	KubeletCert   string `yaml:"kubelet-crt,omitempty"`
+	KubeletKey    string `yaml:"kubelet-key,omitempty"`
+}
+
+func (j *JoinClusterConfig) ToMap() (map[string]string, error) {
```

Comment:
Use `ToMicrocluster()`, so that it better conveys what we do, see also https://github.com/canonical/k8s-snap/blob/37b00e5d0ce172c9dae06078f573be102ebdcfe7/src/k8s/api/v1/bootstrap_config.go#L31-L33